scrunch/datasets.py

```diff
@@ -183,23 +183,23 @@ def get_project(project, connection=None):
     connection = _default_connection(connection)
     sub_project = None
 
-    if '|' in project:
-        project_split = project.split('|')
-        project = project_split.pop(0)
-        sub_project = '|' + '|'.join(project_split)
-
     try:
-        ret = connection.projects.by('name')[project].entity
-    except KeyError:
-        try:
-            ret = connection.projects.by('id')[project].entity
-        except KeyError:
-            raise KeyError("Project (name or id: %s) not found." % project)
-
-    _project = Project(ret)
+        if re.match(r'^[0-9a-f]{32}$', project, re.I):
+            pr_url = '{0}projects/{1}/'.format(connection.session.site_url, project)
```

Comment:
Using shoji you should never be concatenating URLs like this. 

If you have an ID, you should fetch it using:

```
connection.projects.by('id')[project_id].entity
```

Reply:
Indeed, the reason I asked @shaikh-ma to create this PR is because I want a way to get a nested project without having to think navigate to it step-by-step when it's not a root-level folder.

Reply:
I see, you want to fetch any arbitrary ID then yes you're right. You won't be able to get it from the `/projects/` catalog.

In any case, make a different function to get by ID, rather than overloading magic in the existing function. This could very well a `Project.get_by_id(...)` 


If you have the full path of the project, `grand | parent | child` we have some code already in `Project.get()` that achieves this, but it's an instance metod instead of a class method. You could reuse this code to work for both cases. `Project.get_by_path()` and `project_instance.get()`  https://github.com/Crunch-io/scrunch/blob/master/scrunch/datasets.py#L528-L537 - It leverages the existing path parsing code in Scrunch.

Reply:
Thanks, @jjdelc. Good call, we can take the approach of adding a class method `Project.get_by_id(...)` and leave `get_project` alone.